components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -21,6 +21,8 @@ class GeckoEngine(
     private val defaultSettings: Settings? = null
 ) : Engine {
 
+    constructor(context: Context, defaultSettings: Settings? = null) : this(GeckoRuntime.getDefault(context), defaultSettings)
+
```

Comment:
Ah, we can't do that in the existing constructor because we do not have a `Context` here. Mhh, maybe it's not worth adding another constructor for this then. @csadilek what do you think?

Reply:
@pocmo as we just added a `context` parameter to `SystemEngine`, we could now do the same here. I think that would be good, as it would make the API symmetric.

So, we wouldn't need a separate constructor but change the existing one:
```Kotlin
class GeckoEngine(
    context: Context,
    private val runtime: GeckoRuntime = GeckoRuntime.getDefault(context),
    private val defaultSettings: Settings? = null
)
```

Then constructing engines will be the same in both cases:

```Kotlin
engine = SystemEngine(context)
engine = SystemEngine(context, DefaultSettings(...))
engine = GeckoEngine(context)
engine = GeckoEngine(context, DefaultSettings(...))
```

Wdyt?

Reply:
Sounds good!

@fercarcedo Can you update the PR to add this new constructor? :)

Reply:
Updated :)

Reply:
@fercarcedo Thank you! We have code for Gecko build variants in the our sample-browser app (beta|nightly|release) that needs to be updated now though to use this new constructor (pass in the context):

https://github.com/mozilla-mobile/android-components/blob/master/samples/browser/src/geckoBeta/java/org/mozilla/samples/browser/Components.kt#L24

https://github.com/mozilla-mobile/android-components/blob/master/samples/browser/src/geckoNightly/java/org/mozilla/samples/browser/Components.kt#L23

https://github.com/mozilla-mobile/android-components/blob/master/samples/browser/src/geckoRelease/java/org/mozilla/samples/browser/Components.kt#L23

Would you mind updating these as well?

Reply:
Yes, those files were also updated as part of the PR :) (along with the tests)

Reply:
Perfect, thanks!